documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,978 @@
+    public MyTopic(String topic, AWSIotQos qos) {
+        super(topic, qos);
+    }
+
+    @Override
+    public void onSuccess() {
+        // Called when unsubscribing succeeds.
+    }
+    
+        @Override
+    public void onFailure() {
+        // Called when unsubscribing fails.
+    }
+}
+
+// Unsubscribe from topic.
+MyTopic myOwnTopic = new MyTopic("my/topic", AWSIotQos.QOS1);
+client.unsubscribe(myOwnTopic);
+```
+
+**Example of unsubscribing in V2**
+
+```java
+UnsubscribePacketBuilder unsubBuilder = new UnsubscribePacketBuilder("my/topic");
+client.unsubscribe(unsubBuilder.build()).get(60, TimeUnit.SECONDS);
+```
+
+
+### Client Stop
+
+In V1 SDK, the `disconnect` method in the `AWSIotMqttClient` class disconnects the client. Once disconnected, the client
+can connect again by calling `connect`.
+
+In V2 SDK, an MQTT5 client can stop a session by calling the [stop](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/Mqtt5Client.html#stop(software.amazon.awssdk.crt.mqtt5.packets.DisconnectPacket))
+method. You can provide an optional [DisconnectPacket](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/packets/DisconnectPacket.html)
+parameter. A closed client can be started again by calling [start](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/Mqtt5Client.html#start()).
+
+**Example of disconnecting a client in V1**
+
+```java
+client.disconnect();
+```
+
+**Example of disconnecting a client in V2**
+
+```java
+DisconnectPacketBuilder disconnectBuilder = new DisconnectPacketBuilder();
+disconnectBuilder.withReasonCode(DisconnectPacket.DisconnectReasonCode.NORMAL_DISCONNECTION);
+client.stop(disconnectBuilder.build());
+```
+
+
+### Client Shutdown
+
+V1 SDK automatically cleans resources allocated by an `AWSIotMqttClient` object on shutdown.
+
+In V2 SDK, when an MQTT5 client is not needed anymore, your program **must** close it explicitly via a `close` call.
+
+V2 SDK `Mqtt5Client` class implements [AutoCloseable](https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html)
+interface, so it is recommended to create `Mqtt5Client` objects in [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)
+blocks.
+
+**Example of closing a client in V2**
+
+```java
+Mqtt5CLient client = builder.build()
+
+// Once fully finished with the Mqtt5Client:
+client.close();
+```
+
+```java
+// client.close() will be called at the end of the try/catch block.
+try (Mqtt5CLient client = builder.build()) {
+    // ...
+} catch (Exception e) {
+}
+
+```
+
+
+### Reconnects
+
+V1 has a maximum number of retry attempts for auto-reconnect. If you exhausted the maximum number of retries, V1 will throw
+a permanent error and you will not be able to use the same client instance again.
+
+V2 attempts to reconnect automatically until connection succeeds or `client.stop()` is called. The reconnection parameters,
```

Comment:
We probably wanna make it clear that the reconnection only works if the initial start() call succeed. "until connection succeed" might sound like the client would do reconnection even for the initial start() call.

Reply:
It's true for the mqtt311 client only. For the mqtt5 client, if the initial `start()` call fails, the connection status is set to `PENDING_RECONNECT`, I just checked it.

Reply:
Also, I just remembered that we have the following statement in the mqtt5 guide (the "Major Changes" section):
> The MQTT5 client does not treat initial connection failures differently. With the MQTT311 implementation, a failure during initial connect would halt reconnects.